zenoh/src/query.rs

```diff
@@ -77,10 +77,33 @@ impl Default for QueryConsolidation {
 #[non_exhaustive]
 #[derive(Clone, Debug)]
 pub struct Reply {
-    /// The result of this Reply.
-    pub sample: Result<Sample, Value>,
-    /// The id of the zenoh instance that answered this Reply.
-    pub replier_id: ZenohId,
+    pub(crate) sample: Result<Sample, Value>,
+    pub(crate) replier_id: ZenohId,
+}
+
+impl Reply {
+    /// Gets the result of this Reply.
+    pub fn sample(&self) -> Result<&Sample, &Value> {
+        self.sample.as_ref()
+    }
+
+    /// Gets the id of the zenoh instance that answered this Reply.
+    pub fn replier_id(&self) -> ZenohId {
+        self.replier_id
+    }
+
+    /// Attempts to convert reply into sample.
+    pub fn try_into_sample(self) -> Result<Sample, Value> {
```

Comment:
I believe this method is not needed since `impl TryFrom<Reply> for Sample` is already provided.

Reply:
Yes, but it becomes too verbose to write, when compiler requires explicit type annotation like TryInto::<Sample>::try_into(reply) instead of just reply.try_into_sample()

Reply:
Correct, but IMHO it's a proliferation of unnecessary methods. Should we add all `try_into_x` implementations? This specific case would be an exception in the whole codebase.

In general the `try_from` is preferred to `try_into` in the same way implementing `From` is preferred to implementing `Into` (and `From` derives `Into`). Having the user doing `Sample::try_from(reply)` doesn't seem much of a burden.

However, the main problem I have with this is the asymmetry between `.sample()` that returns `Result<&Sample, &Value>` and  `.try_into_sample()` which returns `Result<Sample, Value>`. Should then `.sample()` renamed `.try_sample()`? and `.try_into_sample()` as `.try_sample_into()`?

Reply:
As agreed offline with @DenisBiryukov91 , @sashacmc , and @J-Loudet `.sample()` and `.try_into_sample()` have been renamed to `.result()` and `.into_result()`.